### DISCUSSION TITLE
Respect spacing around colons and assigments

### DISCUSSION THREAD

**Discussion by nuno-andre** [astral-sh/ruff#8862] [Q&A] [answered]
Hi, everyone!

I use to align types and some assignments as follows:

```python3
@dataclass
class Example:
    an_attr:      int
    another_attr: str | None
    whatever:     dict[str, str]


class Colors(IntEnum):
    BLACK  = 0
    RED    = 1
    YELLOW = 2
```

I have searched the documentation and this repo, but I can't find a way to make ruff respect this spacing. Is there any way?

Thanks!

**Comment by zanieb** [comment, marked as answer]
Hi! Our formatter does not support this. There's a similar discussion about alignment of trailing comments at https://github.com/astral-sh/ruff/issues/7684 that you may find interesting. I'd recommend disabling formatting in these sections for now.

There are also some previous discussions over on Black, which we are currently aiming for compatibility with:
- https://github.com/psf/black/issues/1480
- https://github.com/psf/black/issues/3654

**Reply by nuno-andre** [reply]
Thank you for the references! I only meant to ignore formatting around colons and assignments, but yeah, vertical alignment would be a dream.
